### PR TITLE
Add enterprise runner group operations

### PR DESCRIPTION
**Pull request by gabriel-samfira**
This change adds runner group operations on enterprises. The operations are essentially a mirror of the org runner group code, adapted for enterprises.

### CODE REVIEW

File: github/enterprise_actions_runner_groups.go
Comment (gmlewis) [inline github/enterprise_actions_runner_groups.go:36]: Maybe just for consistency?

```suggestion
	TotalCount   *int                      `json:"total_count,omitempty"`
```

Comment (gabriel-samfira) [reply github/enterprise_actions_runner_groups.go]: Done!

**Review by gmlewis — APPROVED** [review]
Thank you, @gabriel-samfira !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by gabriel-samfira** [comment]
I had forgotten about the `omitempty` tag. Added it now.

**Comment by WillAbides** [comment]
For consistency with ActionsService, should the "*RunnerGroup" methods be renamed to "*EnterpriseRunnerGroup"?

This PR also takes the opposite approach with the "*GroupRunners" methods. ActionsService doesn't include "Organization" in those method names, but EnterpriseService includes "Enterprise". I think the ActionsService names make more sense because they are acting on a "Runner", not an "OrganizationRunner" or "EnterpriseRunner".

**Comment by gmlewis** [comment]
> For consistency with ActionsService, should the "*RunnerGroup" methods be renamed to "*EnterpriseRunnerGroup"?
> 
> This PR also takes the opposite approach with the "*GroupRunners" methods. ActionsService doesn't include "Organization" in those method names, but EnterpriseService includes "Enterprise". I think the ActionsService names make more sense because they are acting on a "Runner", not an "OrganizationRunner" or "EnterpriseRunner".

Excellent points, @WillAbides - I think you are right.
@gabriel-samfira - would you mind making these naming changes, please?

**Comment by gabriel-samfira** [comment]
Ohh wow. For some reason I did not get a notification for your comments. I will make these changes ASAP. Thanks for the review!

**Comment by gabriel-samfira** [comment]
Hopefully I got all the names right. Let me know if I missed (or misunderstood) anything.

**Review by WillAbides — APPROVED** [review]

**Comment by ajschmidt8** [comment]
Appreciate the work here! My team will benefit from these changes as well.

Thanks, everyone.

**Pull request merged**
